hack/jenkins/e2e-runner.sh

```diff
@@ -160,7 +160,15 @@ if [[ "${gcp_list_resources}" == "true" ]]; then
   ${gcp_list_resources_script} > "${gcp_resources_before}"
 fi
 if [[ "${E2E_UP,,}" == "true" ]]; then
-    go run ./hack/e2e.go ${E2E_OPT:-} -v --up
+    # We want to try to gather logs even if kube-up fails, so collect the
+    # result here and fail after dumping logs if it's nonzero.
+    go run ./hack/e2e.go ${E2E_OPT:-} -v --up && up_result="$?" || up_result="$?"
+    if [[ -x "cluster/log-dump.sh"  ]]; then
+        ./cluster/log-dump.sh "${ARTIFACTS}"
```

Comment:
I figure, if we're going to make `test/e2e/core.go` point here, then it will just overwrite these logs with new ones once does its thing. If that's not reasonable, then I can make it only log if kube-up fails.